tensorrt_llm/_torch/pyexecutor/py_executor.py

```
@@ -113,6 +113,93 @@ class BatchStatePP(BatchState):
+        """
+        Called when a Cache transceiver or connector transfer is started.
+        1. Increment the counter for the request.
+        2. Releases all resources except for the KV cache, if not already released.
+        3. Store KV cache blocks for reuse.
+        """
+
+        if request.py_request_id in self.requests and should_store_blocks != self.request_should_store_blocks[
+                request.py_request_id]:
+            raise ValueError(
+                f"Request {request.py_request_id} is already in transfer, but should_store_blocks is different"
+            )
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)